torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -68,7 +68,7 @@ class ProcessGroupNCCL : public ProcessGroup {
     bool isSuccess() const override;
 
     // Same as calling synchronize() for NCCL work.
-    bool wait(std::chrono::milliseconds timeout = kUnsetTimeout) override;
+    bool wait(std::chrono::milliseconds timeout = kNoTimeout) override;
```

Comment:
Any reason for using default timeout value for `wait`, but explicitly avoiding introduce an argument to `ProcessGroupNCCL::WorkNCCL::synchronize()`? Will default value also work for `synchronize()`?

Reply:
Briefly chatted with Hongyi about this - we were mainly aiming to change the `wait` API only without many other side-effects on user facing functions, so decided to keep `synchronize` as is. If you think it's better to have a timeout with default value for `synchronize`, we can do that as well.